test/jdk/javax/net/ssl/HttpsURLConnection/SubjectAltNameIPv6.java

```diff
@@ -0,0 +1,344 @@
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+import jdk.test.lib.net.IPSupport;
+import jdk.test.lib.net.SimpleSSLContext;
+import jtreg.SkippedException;
+
+public class SubjectAltNameIPv6 {
+
+    /*
+     * Is the server ready to serve?
+     */
+    volatile static boolean serverReady = false;
+
+    /*
+     * Turn on SSL debugging?
+     */
+    static boolean debug = Boolean.getBoolean("test.debug");
+
+    /*
+     * Define the server side of the test.
+     *
+     * If the server prematurely exits, serverReady will be set to true
+     * to avoid infinite hangs.
+     */
+    void doServerSide() throws Exception {
+        SSLServerSocketFactory sslssf =
+            new SimpleSSLContext().get().getServerSocketFactory();
+        SSLServerSocket sslServerSocket =
+            (SSLServerSocket) sslssf.createServerSocket(serverPort, 0, InetAddress.getByName("[::1]"));
+        sslServerSocket.setEnabledProtocols(new String[]{"TLSv1.3"});
+        serverPort = sslServerSocket.getLocalPort();
+
+        /*
+         * Signal Client, we're ready for his connect.
+         */
+        serverReady = true;
+
+        SSLSocket sslSocket = (SSLSocket) sslServerSocket.accept();
+        OutputStream sslOS = sslSocket.getOutputStream();
+        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(sslOS));
+        bw.write("HTTP/1.1 200 OK\r\n\r\n\r\n");
+        bw.flush();
+        sslSocket.close();
+    }
+
+    /*
+     * Define the client side of the test.
+     *
+     * If the server prematurely exits, serverReady will be set to true
+     * to avoid infinite hangs.
+     */
+    void doClientSide() throws Exception {
+
+        /*
+         * Wait for server to get started.
+         */
+        while (!serverReady) {
+            Thread.sleep(50);
+        }
+
+        SSLSocketFactory sf = new SimpleSSLContext().get().getSocketFactory();
+        URL url = new URL("https://[::1]:" + serverPort + "/index.html");
+        HttpsURLConnection conn = (HttpsURLConnection)url.openConnection();
+
+        /*
+         * Simulate an external JSSE implementation.
+         */
+        conn.setSSLSocketFactory(wrapSocketFactory(sf));
+        conn.getInputStream();
+
+        if (conn.getResponseCode() == -1) {
+            throw new RuntimeException("getResponseCode() returns -1");
+        }
+    }
+
+    // use any free port by default
+    volatile int serverPort = 0;
+
+    // stores an exception thrown by server in a separate thread
+    volatile Exception serverException = null;
+
+    public static void main(String[] args) throws Exception {
+
+        if (!IPSupport.hasIPv6()) {
+            throw new SkippedException("Skipping test - IPv6 is not supported");
+        }
+        if (debug) {
+            System.setProperty("javax.net.debug", "all");
+        }
```

Comment:
If it's the only place where `debug` is used then I believe @myankelev is right and we don't need it.
You can put an `@comment` to remind the reader that they can pass the  `-Djavax.net.debug=all` on the `@run` line.